secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/CompactSerializationJwsSigner.java

```diff
@@ -0,0 +1,198 @@
+import org.forgerock.openig.heap.HeapException;
+import org.forgerock.secrets.Purpose;
+import org.forgerock.secrets.SecretsProvider;
+import org.forgerock.secrets.keys.SigningKey;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.Promises;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link JwsSigner} to sign a {@link Map} message.s<br/>
+ * Provides a Heaplet functionality to be added in the Heap as object for use by the filters and/or the IG scripts<br/>
+ * @see CompactSerializationJwsSigner.Heaplet
+ */
+public class CompactSerializationJwsSigner implements JwsSigner {
+
+    private static final Logger logger = LoggerFactory.getLogger(CompactSerializationJwsSigner.class);
+    private final SigningManager signingManager;
+    private final Purpose<SigningKey> signingKeyPurpose;
+    private final String algorithm;
+    private final String kid;
+
+    public CompactSerializationJwsSigner(
+            SecretsProvider secretsProvider,
+            String signingKeyId,
+            String kid,
+            String algorithm
+    ) {
+        this.signingManager = new SigningManager(secretsProvider);
+        this.kid = kid;
+        this.algorithm = algorithm;
+        this.signingKeyPurpose = Purpose.purpose(signingKeyId, SigningKey.class);
+    }
+
+    /**
+     * Compute the payload signature
+     * <p>
+     *   <ul>
+     *       <li>Accepts a JSON-Based structure payload as a {@link Map} structure</li>
+     *       <p>
+     *          Payload must not be null or empty.
+     *       </p>
+     *       <li>
+     *           Accepts a critical header claims in accordance with
+     *           <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.11'>(RFC7515)"crit" Header Parameter</a>
+     *           as {@link Map}<br/>
+     *           <p>
+     *               Critical header claims can be null or empty, they will be ignored in these cases.
+     *           </p>
+     *       </li>
+     *       <li>
+     *           <p>
+     *               As result, produce a JWS Compact Serialization from JWS, a string composed of three parts encoded in Base64 Url Safe and separated by a dot ( . ), using
+     *               <a href='https://datatracker.ietf.org/doc/html/rfc7159'>JSON-based data structures</a>,
+     *               in accordance with <a href='https://datatracker.ietf.org/doc/html/rfc7515'>RFC7515 JSON Web Signature (JWS)</a>
+     *           </p>
+     *       </li>
+     *   </ul>
+     * </p>
+     * @param payload              {@link Map}
+     * @param criticalHeaderClaims {@link Map}
+     * @return a promise of the resulting object.<br/>
+     * <ul>Promise:
+     *     <li>String: The signed JWT (JSON Web Signature (JWS)) of the task's result</li>
+     *     <li>JwsSignerException: The exception thrown by the task if it fails</li>
+     * </ul>
+     */
+    @Override
+    public Promise<String, JwsSignerException> sign(
```

Comment:
No where do you indicate what the ouput will be. Document in the method description that the alg and kid from the config will be added to the headers and anything else that is automatically added to the resulting JWT

Reply:
added to the documentation